qmi/instruments/newport/smc_100cc.py

```diff
@@ -30,3 +44,542 @@ def __init__(self,
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
+            "FF", friction_compensation))
+        self._check_error()
+        if persist:
+            self.exit_configuration_state(controller_address)
+
+    @rpc_method
+    def get_derivative_gain(self, controller_address: Optional[int] = None) -> float:
+        """
+        Get the derivative gain of the PID control loop.
+
+        Parameters:
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised value of the controller address.
+
+        Returns:
+            The derivative gain of the PID control loop.
+        """
+        _logger.info(
+            "Getting derivative gain of the PID control loop of instrument [%s]", self._name)
+        # instrument must be in configuration state to get the derivative gain of the PID control loop value.
+        self.reset(controller_address)
+        self.enter_configuration_state(controller_address)
+        res = self._scpi_protocol.ask(
+            self._build_command("KD?"))
+        sleep(self.COMMAND_EXEC_TIME)
+        self._check_error()
+        self.exit_configuration_state(controller_address)
+        return float(res[3:])
+
+    @rpc_method
+    def set_derivative_gain(
+            self, derivative_gain: float, persist: bool = False, controller_address: Optional[int] = None) -> None:
+        """
+        Set the derivative gain of the PID control loop.
+
+        Parameters:
+            derivative_gain:    New derivative_gain value in Volt * second/preset unit.
+            persist:            Flag to indicate if the derivative gain should be persisted to the controller's memory,
+                                so it is still available after powering down the controller. When not persisted, the
+                                derivative gain is the one stored in the controller's memory.
+            controller_address: Optional address of the controller that needs to be controlled. By default,
+                                it is set to the initialised derivative_gain of the controller address.
+        """
+        if 1E12 <= derivative_gain or derivative_gain < 0:
+            raise QMI_InstrumentException(
+                f"Provided value {derivative_gain} not in valid range 0 >= derivative_gain > 1E12.")
+
+        _logger.info(
+            "Setting derivative gain of the PID control loop of instrument [%s] to [%f]",
+            self._name, derivative_gain
+        )
+        self.controller_address = controller_address
+        # instrument must be in configuration state to persist the set derivative gain of the PID control loop.
+        if persist:
+            self.reset(controller_address)
+            self.enter_configuration_state(controller_address)
+
+        self._scpi_protocol.write(self._build_command(
```

Comment:
Same ass the `FD` command. It cannot be called in the Ready state. So you would have to make sure the controller is in the Disable state here